src/query/storages/parquet/src/parquet_rs/parquet_reader/reader/full_reader.rs

```diff
@@ -73,13 +80,15 @@ impl ParquetRSFullReader {
         &self,
         loc: &str,
         partition_fields: Option<&[(TableField, Scalar)]>,
-    ) -> Result<ParquetRecordBatchStream<Reader>> {
+    ) -> Result<ParquetRecordBatchStream<ParquetFileReader>> {
         let partition_values_map = partition_fields.map(|arr| {
             arr.iter()
                 .map(|(f, v)| (f.name().to_string(), v.clone()))
                 .collect::<std::collections::HashMap<String, Scalar>>()
         });
+        let meta = self.op.stat(loc).await?;
```

Comment:
this meta is available when listing files?

Reply:
> this meta is available when listing files?

Yes, maybe I can change the code a bit to use existing content length without extra stat call.

Reply:
Fixed by reuse content length in file parts.